tests/conftest.py

```diff
@@ -0,0 +1,350 @@
+
+    Returns:
+        str: The name of the current pytest test.
+    """
+
+    return request.node.name
+
+
+@pytest.fixture(scope="function")
+def get_string() -> Callable:
+    """Return function to get string from file.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _get_string(path: Path) -> str:
+        """Get string from file.
+
+        Args:
+            path: Path to file.
+
+        Returns:
+            String of file
+        """
+
+        if not path.is_file():
+            raise FileNotFoundError(f"File {path} does not exist!")
+
+        with open(path, "r") as file:
+            return file.read()
+
+    return _get_string
+
+
+@pytest.fixture(scope="function")
+def compare_results() -> Callable:
+    """Return function to compare either string or files.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _compare(
+        reference: Union[Path, str],
+        result: Union[Path, str],
+        rtol: float = 1e-14,
+        atol: float = 1e-14,
+        **kwargs,
+    ) -> bool:
+        """Comparison between reference and result with relative or absolute
+        tolerance.
+
+        Args:
+            reference: The reference string or path to the reference file.
+            result: The result string or path to the result file.
+            rtol: The relative tolerance.
+            atol: The absolute tolerance.
+
+        Returns:
+            bool: true if comparison is successful, false otherwise
+        """
+
+        if type(reference) is not type(result):
+            raise TypeError("Reference and result must be of the same type.")
+        elif type(reference) is str and type(result) is str:
+            return compare_strings(reference, result, rtol, atol, **kwargs)
+        elif isinstance(reference, Path) and isinstance(result, Path):
+            if reference.suffix != result.suffix:
+                raise RuntimeError(
+                    "Reference and result file must be of same file type!"
+                )
+            elif reference.suffix in [".vtk", ".vtu"]:
+                return compare_vtk_files(reference, result, rtol, atol)
+            else:
+                raise NotImplementedError(
+                    f"Comparison is not yet implemented for {reference.suffix} files."
+                )
+        else:
+            raise TypeError("Reference and result must be either string or Path.")
+
+    return _compare
+
+
+def compare_strings(
+    reference: str, result: str, rtol: float, atol: float, string_splitter=" "
+) -> bool:
+    """Compare if two strings are identical within a given tolerance. If no
+    tolerance is given strings are compared for equality.
+
+    Args:
+        reference: The reference string.
+        result: The result string.
+        rtol: The relative tolerance.
+        atol: The absolute tolerance.
+        string_splitter: With which string the strings are split.
```

Comment:
I really like the "new" docstring format when using type hints, that we don't have to specify the type here.